packages/core/test/predictableExec.test.ts

```diff
@@ -624,4 +624,43 @@ describe('predictableExec', () => {
 
     expect(received).toEqual({ type: 'KNOCK_KNOCK' });
   });
+
+  it('parent should be able to read the updated state of a child when receiving an event from it', (done) => {
+    let shouldCheck = false;
+
+    const child = createMachine({
+      predictableActionArguments: true,
+      initial: 'a',
+      states: {
+        a: {
+          after: {
+            1: 'b'
+          }
+        },
+        b: {
+          entry: [
+            () => (shouldCheck = true),
+            sendParent({ type: 'CHILD_UPDATED' })
+          ]
+        }
+      }
+    });
+
+    const machine = createMachine({
+      predictableActionArguments: true,
+      invoke: {
+        id: 'myChild',
+        src: child
+      }
+    });
+
+    const service = interpret(machine).start();
+
+    service.subscribe((state) => {
+      if (shouldCheck) {
```

Comment:
Nit: I would add a `pass` and `fail` state to the `machine` and have a guarded transition on `CHILD_UPDATED` that goes to `pass` or `fail` depending on the snapshot value of the child, but that can be done as a refactor

Reply:
changed